### PR TITLE
[NVPTX] Make nvptx mma instructions convergent.

### PR DESCRIPTION
**Pull request by weiweichen**
We are running into NVPTX backend generating wrong code for an input:
```
%0 = llvm.nvvm.mma.m?n?k?.row.col.??? (...)
if laneid == 0:
  ret
else:
  store %0
```

The backend reorder the instruction (as an effect of `MachineSink` pass) to
```
if laneid == 0:
  ret
else:
  %0 = llvm.nvvm.mma.m?n?k?.row.col.??? (...)
  store %0
```

This is incorrect because `mma` is a warp instruction which needs all threads to sync before performing the operation instead of being guarded by a specific thread id. It should be similar as the shuffle instruction `shfl` in terms of warp level sync, and `shfl` is marked as `isConvergent = true`.

Apply `isConvergent = true` to `mma` instructions.

### CODE REVIEW

**Review by Mogball — APPROVED** [review]

**Review by Mogball — COMMENTED** [review]
This makes sense to me. These instructions can't be sunk across conditional boundaries. Please make sure to get a review from someone who normally touches the NVPTX backend!

File: llvm/test/CodeGen/NVPTX/mma-no-sink-after-laneid-check.ll [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+declare noundef i32 @llvm.nvvm.read.ptx.sreg.tid.x() #1
+
+declare { float, float, float, float } @llvm.nvvm.mma.m16n8k4.row.col.tf32(i32, i32, i32, float, float, float, float) #2
+
+declare noundef i32 @llvm.nvvm.read.ptx.sreg.laneid() #1
+
+; COM: llvm.nvvm.mma should not sink to the next block and gets reordered to be after laneid check.
+; CHECK-LABEL: no_reorder_mma_and_laneid_check
+define dso_local void @no_reorder_mma_and_laneid_check(ptr %0, ptr %1, i64 %2) #0 {
```
Comment (qcolombet) [inline llvm/test/CodeGen/NVPTX/mma-no-sink-after-laneid-check.ll:15]: Please get rid of the implicit variables (run `opt -passes=instnamer` on your input IR and update the file)

**Review by qcolombet — APPROVED** [review]
LGTM.

`isConvergent` is indeed missing on these instructions.

Nice finding!

File: llvm/test/CodeGen/NVPTX/mma-no-sink-after-laneid-check.ll (hunk 1 above)
Comment (weiweichen) [reply llvm/test/CodeGen/NVPTX/mma-no-sink-after-laneid-check.ll]: Yep, update! Thank you for the `opt` tip!

**Pull request merged**